rotary/rules/rotary.smk

```diff
@@ -98,6 +110,13 @@ rule download_contamination_reference:
         fi
         """
 
+rule set_up_custom_contamination_references:
+    output:
+        expand(os.path.join(DB_DIR_PATH, 'contamination_references', 'custom', '{contaminant_name}.fna.gz'),
+            contaminant_name=CUSTOM_CONTAMINATION_FILE_NAMES),
+    run:
+        for name, path in zip(CUSTOM_CONTAMINATION_FILE_NAMES,CUSTOM_CONTAMINATION_FILEPATHS):
+            symlink_or_compress(path, os.path.join(DB_DIR_PATH, 'contamination_references', 'custom', f'{name}.fna.gz'))
```

Comment:
This function assumes that the custom FastA files are gzipped. Do we want to require that the custom FastA files are gzipped? I think the BBDuk ref argument can handle FastA or gzipped FastA.

After checking if the input files are gzipped, we will either need to:
- throw an error if the files are not gzipped, 
- compress non-gzipped files, or
- handle their filepaths differently than gzipped files during symlinking and downstream steps

Reply:
@jmtsuji Isn't that what `symlink_or_compress` function does? If the file is already compressed, it symlinks it. It compresses and writes the compressed output to the desired path if it is not compressed. That was the function's intended purpose. Or am I missing something?

Reply:
@jmtsuji, see this function and the function above it. Python can gzip files using the zlib library, the underlying code for gzip.

https://github.com/jmtsuji/rotary/blob/c6ff6557d46609d5fd28edf005c4bddaa22218f0/rotary/utils.py#L91

Reply:
@LeeBergstrand Sorry, this was totally an oversight on my end. I think I must have read this code too hastily and somehow missed `symlink_or_compress`. Yes, your previous function addresses the gzip issue, so the code is OK as-is. My apologies for the wild goose chase!